fplll/bkz.cpp

```diff
@@ -803,11 +803,107 @@ template <class ZT, class FT> bool BKZAutoAbort<ZT, FT>::test_abort(double scale
   return no_dec >= maxNoDec;
 }
 
+// call pruner
+template <class FT>
+int bkz_prune_f(ZZ_mat<mpz_t> &b, const BKZParam &param, int sel_ft, double lll_delta,
```

Comment:
I don't see why this function is called `bkz_` or why it is in `bkz.cpp`, it just calls the pruner, right?

Reply:
Yes Martin. One reason is that the pruner can do a BKZ-like task: optimizing the pruning coefficients for block [i, j]. Also I want to re-use the bkz_reduction() function which filters the floating point types (to avoid to have another function do the same task).

Reply:
I'd rather say that BKZ also does a Pruning task, so I wouldn't put the pruning stuff in BKZ :) For the second reason, I'd re-factor and move the FP filtering into its own subfunction, that can be re-used

Reply:
That sounds a good solution and we could move the bkz_prune_f to pruner.cpp. Let me take care of this.

Reply:
Ta!

Reply:
The command line stuff is factored out of bkz.cpp into pruner.cpp. The bkz.cpp and bkz.h are not touched in the pull request now.